add community-membership.md (#11)
tikv/community: community-membership.md

# Community Membership

## Maintainer

Maintainers are the planners and designers of the project, with the authority to merge branches into the master. Maintainers come from Committers. They must demonstrate both good judgment and responsibility towards the health of the subproject. Maintainers must set up the technical direction and make or approve design decisions for the subproject - either directly or by delegating these responsibilities.

### How to become a Maintainer:

- Demonstrated a deep understanding of TiKV's technical goals and directions
- Actively organize technical seminars and propose constructive projects
- Can handle complex problems in the code implementation process

> - Obtained unanimous votes from Members of the PMC

Reviewer: ```suggestion
- Obtained unanimous votes from the PMC
```

### Privileges and responsibilities:

- Develop technical design and decisions for subprojects
- Set the technical directions and priorities for subprojects
- Participate in product releases and roadmap development of the project
- Continuously contribute to the project by mentoring Contributors, Reviewers, and Committers
- Ensure the overall quality of the subprojects

## Committer

> Committers are recommended by Maintainers or the PMC as Contributors with outstanding contributions to TiKV. A committer must have completed at least one feature or fix a major bug independently.

Reviewer: ```suggestion
Committers are recommended by Maintainers or the PMC as Contributors with outstanding contributions to TiKV. A Committer must have completed at least one feature or fix a major bug independently.
```

### How to become a Committer:

- Demonstrated good technical judgment
- Assumed the Reviewer role in at least two or more subprojects
- Reviewed or merged a certain amount of TiKV code
- Contributed significantly to TiKV performance improvements, or fix major bugs
- Voted by two or more incumbent Maintainers

### Privileges and responsibilities:

- Control overall code quality of the project
- Guide Contributors and Reviewers to contribute to TiKV continuously
- Participate in Committer meetings on a regular basis and have knowledge priority regarding project development

## Reviewer

Reviewers are responsible for reviewing the code submitted by the community to ensure the quality and correctness; generated out of Contributors.

### How to become a Reviewer:

- Familiar with the codebase
- Contributed a certain amount of code to TiKV subprojects

> - Performed excellently in the Special Interest Group (SIG)

Reviewer: ```suggestion
- Performed excellently in a Special Interest Group (SIG)
```

# Community Membership

## Maintainer

> Maintainers are the planners and designers of the project, with the authority to merge branches into the master. Maintainers come from Committers. They must demonstrate good judgment and responsibility for the health of the subproject. The maintainer must set up the technical direction directly or by delegating these responsibilities, and make or approve design decisions for the subproject.

Reviewer: ```suggestion
Maintainers are the planners and designers of the project, with the authority to merge branches into the master. Maintainers come from Committers. They must demonstrate both good judgment and responsibility towards the health of the subproject. Maintainers must set up the technical direction and make or approve design decisions for the subproject - either directly or by delegating these responsibilities.
```

### How to become a Maintainer:

> - Demonstrate a deep understanding of TiDB's technical goals and directions

Reviewer 2: TiKV's

> - Demonstrate a deep understanding of TiKV's technical goals and directions

Reviewer: ```suggestion
- Demonstrated a deep understanding of TiKV's technical goals and directions
```

> - Actively organize technical seminars and propose constructive projects

Reviewer: This sounds more like a responsibility...

Author: The Maintainer need to show their ability first, by deeply involved in TiKV and propose constructive projects

Reviewer: Then maybe  "Actively involved in organizing technical seminars and proposing constructive subprojects"?

Author: done

- Can handle complex problems in the code implementation process

> - Get unanimous votes from Members of the PMC

Reviewer: ```suggestion
- Obtained unanimous votes from Members of the PMC
```

### Privileges and responsibilities:

- Develop technical design and decisions for subprojects
- Set the technical directions and priorities for subprojects
- Participate in product releases and roadmap development of the project
- Continuously contribute to the project by mentoring Contributors, Reviewers, and Committers
- Ensure the overall quality of the subprojects

## Committer

Committers are recommended by Maintainers or the PMC as Contributors with outstanding contributions to TiKV. A committer must have completed at least one feature or fix a major bug independently.

### How to become a Committer:

- Demonstrated good technical judgment
- Assumed the Reviewer role in at least two or more subprojects

> - Reviewed or merged a certain amount of TiKV code

Reviewer: How much exactly... I see K8s has an exact number @winkyao @gingerkidney

Author: It depends on different module

- Contributed significantly to TiKV performance improvements, or fix major bugs
- Voted by two or more incumbent Maintainers

### Privileges and responsibilities:

- Control overall code quality of the project
- Guide Contributors and Reviewers to contribute to TiKV continuously
- Participate in Committer meetings on a regular basis and have knowledge priority regarding project development

## Reviewer

Reviewers are responsible for reviewing the code submitted by the community to ensure the quality and correctness; generated out of Contributors.

### How to become a Reviewer:

- Familiar with the codebase

> - Contributed a certain amount of code to TiKV subprojects

Reviewer: how much?

Author: ditto

- Performed excellently in the Special Interest Group (SIG)
- Sponsored by at least two incumbent Committers

### Privileges and responsibilities:

- Ensure the correctness of the subproject code

> 

Reviewer: Why blank line in-between?

Author: fixed

- Respond promptly to PRs submitted by the community

- Participate in SIG seminars on a regular basis

- Approve code reviews with LGTM (Looks Good To Me)

## Active Contributor

Active community contributors.

### How to become a Reviewer:

- Contributed 8 PRs and above to TiKV within one year

- Received recognition from two incumbent Reviewers

### Privileges and responsibilities:

- Automatically qualify for entry to the Special Interest Group (SIG)

## Contributor

> - Merged at least 1 PR in TiKV Repo

Reviewer: ```suggestion
- Merged at least 1 PR in any Repo under [TiKV](https://github.com/tikv) 
```
